File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/course/CourseSubcomponent.java
package com.tevinjeffrey.rutgersct.ui.course;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent
public interface CourseSubcomponent
    extends AndroidInjector<CourseFragment> {

  void inject(CourseViewModel coursePresenter);

  @Subcomponent.Builder
  abstract class Builder extends AndroidInjector.Builder<CourseFragment> {
    @Override public abstract CourseSubcomponent build();
  }
}

File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/course/CourseSubcomponent.kt
package com.tevinjeffrey.rutgersct.ui.course

import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent
interface CourseSubcomponent : AndroidInjector<CourseFragment> {

  fun inject(coursePresenter: CourseViewModel)

  @Subcomponent.Builder
  abstract class Builder : AndroidInjector.Builder<CourseFragment>() {
    abstract override fun build(): CourseSubcomponent
  }
}
